analytics.js

```diff
@@ -176,26 +176,27 @@ class Analytics {
         this.loadOnlyIntegrations,
         this.clientIntegrations
       );
-      let cookieConsent = undefined;
       // Check if cookie consent manager is being set through load options
       if (Object.keys(this.cookieConsentOptions).length) {
         // Call the cookie consent factory to initialise and return the type of cookie
         // consent being set. For now we only support OneTrust.
-        cookieConsent = CookieConsentFactory.initialize(
-          response,
-          this.cookieConsentOptions
-        );
+        try {
+          let cookieConsent = CookieConsentFactory.initialize(
+            this.cookieConsentOptions
+          );
+          // If cookie consent object is return we filter according to consents given by user
+          // else we do not consider any filtering for cookie consent.
+          this.clientIntegrations = this.clientIntegrations.filter((intg) => {
+            return (
+              !cookieConsent || // check if cookieconsent object is present and then do filtering
+              (cookieConsent && cookieConsent.isEnabled(intg.config))
+            );
+          });
+        } catch (e) {
+          logger.error(e);
+        }
```

Comment:
Why do we have to put the entire thing under try-catch? Aren't we just concerned about initializing the consent object

Reply:
Then even if it goes to catch it will be executed

Reply:
Got it